crates/ty_python_semantic/src/types.rs

```diff
@@ -9197,6 +9234,56 @@ impl<'db> BoundSuperType<'db> {
     }
 }
 
+#[salsa::interned(debug)]
+pub struct TypeIsType<'db> {
+    #[return_ref]
+    ty: Type<'db>,
```

Comment:
nit: would prefer a more descriptive name here, e.g. `narrowed_type`. (We've also been moving away from using variables named `ty`)